Fix gizmo size in 2D (ptOrthographic)

and3md opened the pull request:
In 2D gizmos can be really big. I changed algorithm to set scale about 15% of effective height. I also make them not grow/decrease when scale is changed - I think its more handy. Tested on platformer and 2d game template.

Before:

![obraz](https://user-images.githubusercontent.com/18555708/114381693-aff8a700-9b8b-11eb-905b-74d462514fff.png)


After:
![obraz](https://user-images.githubusercontent.com/18555708/114381513-7c1d8180-9b8b-11eb-8412-2536cefb5dda.png)

![obraz](https://user-images.githubusercontent.com/18555708/114381613-9bb4aa00-9b8b-11eb-970d-14049d8f33ce.png)

michaliskambi commented:
I simplified the code a little more, preserving your logic -- in case of `Camera.ProjectionType = ptOrthographic`, most of the existing calculations do not have to be done to get `ScaleUniform`. I extracted them to local function `GetPerspectiveGizmoScale`.

Please test, the effect (on your platformer game too) shoud be as if after your PR.

I can also reproduce the problem with grabbing dragon/trees you mentioned on Friday -- have it in TODO to solve ASAP.

michaliskambi commented:
> I can also reproduce the problem with grabbing dragon/trees you mentioned on Friday -- have it in TODO to solve ASAP.

Solved.

and3md commented:
> I simplified the code a little more, preserving your logic -- in case of `Camera.ProjectionType = ptOrthographic`, most of the existing calculations do not have to be done to get `ScaleUniform`. I extracted them to local function `GetPerspectiveGizmoScale`.
> 
> Please test, the effect (on your platformer game too) should be as if after your PR.

You are right, works great! Thanks!

and3md commented:
> > I can also reproduce the problem with grabbing dragon/trees you mentioned on Friday -- have it in TODO to solve ASAP.
> 
> Solved.

I still can reproduce that I can't get blue arrow with https://github.com/castle-engine/castle-engine/pull/279 merged on current master (dragon 2d template). Maybe there is something else to fix.

michaliskambi commented:
> I still can reproduce that I can't get blue arrow with #279 merged on current master (dragon 2d template). Maybe there is something else to fix.

Hm, testing with latest master (without any extra PRs) the problem seems fixed for me. I mean, I create new project "2d game" and can reliably select and move dragon/trees by dragging the gizmo. I didn't yet try with https://github.com/castle-engine/castle-engine/pull/279 .
